feat: created profile page

kobekobe0 opened the pull request:
overview tab
edit tab
files tab

kobekobe0 committed:
feat: created profile page

overview tab
edit tab
files tab

stevengragg commented on lib/actions/profile-actions.ts:
```diff
@@ -0,0 +1,52 @@
+import * as z from "zod";
+
```
Do we still need this or the `services/user/update-profile.ts` ?

kobekobe0 replied:
will remove this po

stevengragg commented on app/(applicant)/app/profile/files/page.tsx:
```diff
@@ -0,0 +1,87 @@
+import FileUpload from "@/components/profile/file-upload";
+import { fileUploadSchema, onSubmit, FileUploadSchema } from "@/services/user/update-files";
+
+const FileUploadForm: React.FC = () => {
```
Please refrain from using React.FC because every react component are implicitly returning JSX elements and in the case of Props, we can do normal type definition using Type. If we need to extends or alter the props in some parts we can do that by using Interface and extending the type definition.

kobekobe0 replied:
changed

stevengragg commented on types/file-upload.ts:
```diff
@@ -0,0 +1,9 @@
+
+export interface FileUploadProps {
```
If this is a normal props of a react component define this inside that react component but if this needs elsewhere and the type definition of the props needs to be extended or altered in any way, then use Interface. Otherwise, use type and declare it inside the React component.

kobekobe0 replied:
removed this

stevengragg replied:
So we are talking about `FileUploadProps` and  `IProfileProps` here. If you only need each in a specific React component, then declare it on that file.

stevengragg commented on types/profile.ts:
```diff
@@ -0,0 +1,25 @@
+export interface IProfileProps {
```
Same comment https://github.com/stevengragg/access-virtual-staffing/pull/22/files#r1954535529

kobekobe0 replied:
changed

stevengragg commented on services/user/update-profile.ts:
```diff
@@ -0,0 +1,107 @@
+
+export type EditProfileSchema = z.infer<typeof editProfileSchema>;
+
+export const onSubmit = (data: EditProfileSchema) => {
```
Just for my preference, please do not export onSubmit function but create a handler function in this case:

// Inside the React component
```typescript
export function ProfileForm() {
 //..
      function onSubmit (data: EditProfileSchema) {
              handleProfileUpdate(data)
          //...
       }
   //..
```
// Inside the action file
```typescript
export const handleProfileUpdate = (data: EditProfileSchema) ...
```

kobekobe0 replied:
didn't notice, onsubmit on update-profile isnt being used, removed it

moved onSubmit from services to component in file-upload.ts

stevengragg commented on app/(applicant)/app/profile/page.tsx:
```diff
@@ -1,26 +1,32 @@
-  },
-  { returnTo: "/app/profile" }
-);
+export default function ProfilePage() {
```
Do not remove the `withPageAuthRequired` authentication wrapper.

```typescript
export default withPageAuthRequired(
  async function Profile() {
  ..//
  },
  { returnTo: "/app/profile" }
  );
```

kobekobe0 replied:
added

stevengragg commented on components/profile/file-upload.tsx:
```diff
@@ -0,0 +1,70 @@
+import { X } from "lucide-react";
+import {FileUploadProps} from "@/types/file-upload";
+
+const FileUpload: React.FC<FileUploadProps> = ({
```
Same comment https://github.com/stevengragg/access-virtual-staffing/pull/22/files#r1954532665

kobekobe0 replied:
changed

stevengragg commented on components/profile/profile-overview-card.tsx:
```diff
@@ -0,0 +1,156 @@
+import { useUser } from "@auth0/nextjs-auth0/client";
+import {IProfileProps} from "@/types/profile";
+
+const ProfileCard: React.FC<IProfileProps> = ({ profile }) => {
```
Same comment https://github.com/stevengragg/access-virtual-staffing/pull/22/files#r1954532665

kobekobe0 replied:
changed

stevengragg commented on components/profile/tabs.tsx:
```diff
@@ -0,0 +1,41 @@
+  { name: "Files", path: "/app/profile/files" },
+];
+
+const ProfileTabs: FC = () => {
```
Same comment https://github.com/stevengragg/access-virtual-staffing/pull/22/files#r1954532665

kobekobe0 replied:
changed

stevengragg commented on app/(applicant)/app/profile/layout.tsx:
```diff
@@ -0,0 +1,45 @@
+  children: ReactNode;
+}
+
+export default function DashboardLayout({ children }: LayoutProps) {
```
Why do we call this DashboardLayout?

kobekobe0 replied:
changed

kobekobe0 committed:
fix: profile page
- renamed DashboardLayout to EditProfileLayout
- added withPageAuthRequired
- change interface to types on profile-overview and edit-profile
- removed onsubmits on services and putted inside component itself

kobekobe0 committed:
fix:

fixed layout
fixed dropdown bancground color

note: couldnt find a way to display error message of fields that are array cause error is depented on the fieldstate
